core/liquidity/supplied/statevar.go

```diff
@@ -97,66 +93,58 @@ func (e *Engine) startCalcProbOfTrading(eventID string, endOfCalcCallback statev
 	endOfCalcCallback.CalculationFinished(eventID, res, nil)
 }
 
-// calculateBidRange calculates the probabilities of price between bestBid and the worst bid (minBid)
-// in increments of <tickSize> and records the offsets and probabilities
+// calculateBidRange calculates the probabilities of price between bestBid and the worst bid
+// in increments of incrementFactor of the best bid and records the offsets and probabilities
 // such that offset 0 is stored in offsets[0] which corresponds to the best bid
 // and similarly probabilities[0] corresponds to best bid trading probability
 // whereas the last entry in offset equals to the maximum distance from bid for which
-// probability is calculated, and the last entry in probabilities corresponds to
-// the probability of trading at the price implied by this offset from best bid.
-func calculateBidRange(bestBid, minBid, tickSize, tauScaled num.Decimal, probabilityFunc func(num.Decimal, num.Decimal, num.Decimal, num.Decimal, num.Decimal, bool, bool) num.Decimal) ([]num.Decimal, []num.Decimal) {
-	offsets := make([]num.Decimal, 0, int(numberOfPricePoints.Uint64()))
-	probabilities := make([]num.Decimal, 0, int(numberOfPricePoints.Uint64()))
+// probability is calculated and is greater than the default minimum acceptable probability of trading.
+func calculateBidRange(bestBid, incrementFactor, tauScaled num.Decimal, probabilityFunc func(num.Decimal, num.Decimal, num.Decimal, num.Decimal, num.Decimal, bool, bool) num.Decimal) ([]num.Decimal, []num.Decimal) {
+	offsets := []num.Decimal{}
+	probabilities := []num.Decimal{}
 
 	p := bestBid
 	offset := num.DecimalZero()
-	for p.GreaterThan(minBid) && !p.IsNegative() {
+	increment := incrementFactor.Mul(bestBid)
+	stop := false
+	for !stop {
 		offsets = append(offsets, offset)
-		prob := probabilityFunc(bestBid, p, minBid, bestBid, tauScaled, true, true)
+		prob := probabilityFunc(bestBid, p, num.DecimalZero(), bestBid, tauScaled, true, true)
 		probabilities = append(probabilities, prob)
-		if p.Equal(minBid) {
-			break
+		if prob.LessThanOrEqual(defaultMinimumProbabilityOfTrading) {
```

Comment:
The comment of this function says:
```
// whereas the last entry in offset equals to the maximum distance from bid for which
// probability is calculated and is greater than the default minimum acceptable probability of trading.
```
but given we've already added to the slices before the check, the last entry will be less-than-or-equal to `defaultMinimumProbabilityOfTrading`, not greater than.

Reply:
I don't think this has been fixed. On line 113 we add to the slice`probabilities = append(probabilities, prob)` and then on 114 we check and stop if its less `defaultMinimumProbabilityOfTrading`, but by then its already in the slices. Line 114 needs to come before we call `append()` on `probabilities` and `offsets`.